### PR TITLE
doc: Convert builders/images/* to markdown

### LINKED ISSUE #105758: Convert appimagetools from Docbook to CommonMark
This issue tracks the conversion of `/doc/builders/images/appimagetools.xml` from Docbook to CommonMark per [RFC 72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).

If you intend to work on this conversion, please comment below!

# Conversion instructions

1. Comment here that you intend to work on it. (Please also post if you stop working on it.)
2. Checkout a new branch of nixpkgs master.
3. Edit `doc/builders/images.xml`: change `appimagetools.xml` to `appimagetools.section.xml`.
4. Rename `doc/builders/images/appimagetools.xml` to `appimagetools.section.md`.
5. Convert the contents from Docbook to [CommonMark](https://commonmark.org/). Follow [conversion best practices](https://github.com/NixOS/nixpkgs/issues/105243). Use existing `.md` sections for inspiration.
6. Use `nix-build` from the `doc/` folder to build the manual.
7. Preview the manual on your computer at `http://0.0.0.0:8000/manual.html` after running from `doc/` folder:
  ```
  nix-shell -p python3 --run 'python3 -m http.server --directory ./result/share/doc/nixpkgs'
  ```
8. When it looks good, submit a PR and reference this issue.

### LINKED ISSUE #105761: Convert snaptools from Docbook to CommonMark
This issue tracks the conversion of `/doc/builders/images/snaptools.xml` from Docbook to CommonMark per [RFC 72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).

If you intend to work on this conversion, please comment below!

# Conversion instructions

1. Comment here that you intend to work on it. (Please also post if you stop working on it.)
2. Checkout a new branch of nixpkgs master.
3. Edit `doc/builders/images.xml`: change `snaptools.xml` to `snaptools.section.xml`.
4. Rename `doc/builders/images/snaptools.xml` to `snaptools.section.md`.
5. Convert the contents from Docbook to [CommonMark](https://commonmark.org/). Follow [conversion best practices](https://github.com/NixOS/nixpkgs/issues/105243). Use existing `.md` sections for inspiration.
6. Use `nix-build` from the `doc/` folder to build the manual.
7. Preview the manual on your computer at `http://0.0.0.0:8000/manual.html` after running from `doc/` folder:
  ```
  nix-shell -p python3 --run 'python3 -m http.server --directory ./result/share/doc/nixpkgs'
  ```
8. When it looks good, submit a PR and reference this issue.

### PR DESCRIPTION
**Pull request by bobby285271**
###### Motivation for this change

Closes #105758 and closes #105761

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Converted xml to md and updated images.xml to reference it.

### CODE REVIEW

File: doc/builders/images/appimagetools.section.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+type2.AppImage: ELF 64-bit LSB executable, x86-64, version 1 (SYSV) (Lepton 3.x), scale 232-60668, spot sensor temperature -4.187500, color scheme 15, show scale bar, calibration: offset -0.000000, slope 0.000000 (Lepton 2.x), scale 4111-45000, spot sensor temperature 412442.250000, color scheme 3, minimum point enabled, calibration: offset -75402534979642766821519867692934234112.000000, slope 5815371847733706829839455140374904832.000000, dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.18, BuildID[sha1]=79dcc4e55a61c293c5e19edbd8d65b202842579f, stripped\012- data
+```
+
+Note how the type 1 AppImage is described as an `ISO 9660 CD-ROM filesystem`, and the type 2 AppImage is not.
+
+## Wrapping {#ssec-pkgs-appimageTools-wrapping}
+
+Depending on the type of AppImage you're wrapping, you'll have to use `wrapType1` or `wrapType2`.
+
+```{=docbook}
```
Comment (SuperSandro2000) [inline doc/builders/images/appimagetools.section.md:33]: Can you convert this, too?

Comment (bobby285271) [reply doc/builders/images/appimagetools.section.md]: Done, please take a look!

**Review by jtojnar — APPROVED** [review]
Diff looks okay to me.

**Comment by bobby285271** [comment]
Added converts of `snaptools.xml`.

I changed the title of section 14.4.3 and move the examples into the the markdown file during converting, so please take a look!

Thanks in advance for reviewing :-)

File: doc/builders/images/snaptools.section.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  };
+}
+```
+
+`nix-build` this expression and install it with `snap install ./result --dangerous`. `hello` will now be the Snapcraft version of the package.
+
+## Build a Graphical Snap {#ssec-pkgs-snapTools-build-a-snap-firefox}
+
+Graphical programs require many more integrations with the host. This example uses Firefox as an example, because it is one of the most complicated programs we could package.
+
```
Comment (jtojnar) [inline doc/builders/images/snaptools.section.md:37]: Maybe we should add the example anchors as inline anchors:

```suggestion

[]{#ex-snapTools-buildSnap-firefox}
```

File: doc/builders/images/snaptools.section.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+```nix
```
Comment (jtojnar) [inline doc/builders/images/snaptools.section.md:38]: Or mark the code block directly.

```suggestion
``` {#ex-snapTools-buildSnap-firefox .nix}
```

**Review by jtojnar — COMMENTED** [review]
Otherwise, the snap parts look fine as well.

**Review by jtojnar — APPROVED** [review]
The diff looks good to me.

**Comment by jtojnar** [comment]
HTML looks okay too.

**Pull request merged**